CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+# SPDX-FileCopyrightText: 2009 Cristian Oneț <onet.cristian@gmail.com>
+
+project(ofx)
+
+set (VERSION_MAJOR 0)
+set (VERSION_MINOR 10)
+set (VERSION_PATCH 5)
+set (VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
+
+cmake_minimum_required(VERSION 2.6)
+
+list(APPEND CMAKE_MODULE_PATH  ${CMAKE_SOURCE_DIR})
+find_package(OpenSP REQUIRED)
+
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif(MSVC)
+
+include_directories(
+    inc
+    ${OPENSP_INCLUDES}
+    ${CMAKE_INSTALL_PREFIX}/include
+)
+
+add_definitions(
+    -D__WIN32__
+    -DMAKEFILE_DTD_PATH="${CMAKE_PREFIX_PATH}/share/libofx/dtd/"
+)
+
+file(GLOB COEFF_HDRS inc/libofx.h lib/*.hh )
+file(GLOB SRC_SOURCES lib/*.cpp ${COEFF_HDRS})
+
+
+# We would like it to be shared, but libofx does not export symbols manually
+# thus it causes troubles on MSVC
+#add_library(ofx SHARED ${SRC_SOURCES})
+add_library(ofx ${SRC_SOURCES})
+
+FIND_LIBRARY( ICONV_LIBRARIES iconv )
+if(MSVC)
+    set_target_properties(ofx PROPERTIES OUTPUT_NAME "libofx")
+endif(MSVC)
+target_link_libraries(ofx ${OPENSP_LIBRARIES} ${ICONV_LIBRARIES})
+
+######### add a utility function so that we can test ofx files ##########
+set(ofxdump_SRCS
+    ofxdump/cmdline.c
+    ofxdump/ofxdump.cpp 
+)
+
+if(MSVC)
+    set(ofxdump_SRCS ${ofxdump_SRCS} lib/messages.cpp lib/getopt.c lib/getopt1.c)
+endif(MSVC)
+
+add_definitions(-DCMDLINE_PARSER_PACKAGE=\"ofxdump\" -DCMDLINE_PARSER_PACKAGE_NAME=\"ofxdump\" -DCMDLINE_PARSER_VERSION=\"${VERSION}\")
```

Comment:
Also a leftover from the original file. I haven't gotten to actually understand the purpose behind these, so left as is for now.

Reply:
`add_definitions` just pastes the argument string into the compiler invocation, which is why you need to escape the quotes so that cmake doesn't eat them. `target_compile_definitions` does a little preprocessing for you so you don't need to escape the quotes or provide the `-D`; it also applies the arguments only to the specified target instead of globally to all targets the way `add_definitions` does.

IOW it's telling the build system to say `gcc -c -DCMDLINE_PARSER_PACKAGE="ofxdump" -o target.o -c source.c`.